citre-lang-c.el

```diff
@@ -67,13 +67,32 @@
       ;; referenced differently in different files.  We only keep the
       ;; non-directory part.  By doing so we can't match #includes that uses
       ;; paths, but we make use of binary search of readtags.
-      (let ((symbol (file-name-nondirectory
-                     (buffer-substring-no-properties
-                      (car bounds) (cdr bounds)))))
+      (let* ((xsymbol (buffer-substring-no-properties
+                       (car bounds) (cdr bounds)))
+             (symbol (file-name-nondirectory xsymbol))
+             ;; Remove upper components above ../
+             ;; ../linux/foo.h => linux/foo.h
+             (suffix (let* ((rstr (reverse xsymbol))
+                            (end (string-match "/\\.\\." rstr))
+                            (substr (if end
+                                        (reverse (substring rstr 0 end))
+                                      xsymbol)))
```

Comment:
Can we do this in a more straightforward way? like

``` elisp
(progn
  (string-match "\\(?:\\.\\./\\)*\\(.*\\)" "../linux/foo.h")
  (match-string 1 "../linux/foo.h"))
;; => "linux/foo.h"
```

since `citre-lang-c` requres `rx`, you can write the regexp like

``` elisp
(rx (* "../") (group (* anychar)))
```